src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
     (ActorFunc)EnEncount3_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_060009A0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD058.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD084.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD194.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/func_809AD1EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Encount3/EnEncount3_Draw.s")
+s32 D_809AD810 = false;
+
+void EnEncount3_Init(Actor* thisx, PlayState* play) {
+    EnEncount3* this = THIS;
+
+    this->unk14A = ENCOUNT3_GET_SPAWN_INDEX(thisx);
+    this->unk158 = ENCOUNT3_GET_PARAM_F80(thisx);
+    this->unk15C = ENCOUNT3_GET_PARAM_7F(thisx);
+    this->unk15A = this->actor.csId;
+    this->unk16C = this->actor.world.rot.z * 10.0f;
+    if (this->unk16C < 10.0f) {
+        this->unk16C = 10.0f;
+    } else if (this->unk16C > 1000.0f) {
+        this->unk16C = 1000.0f;
+    }
+    if (this->unk15C == 0x7F) {
+        this->unk15C = -1;
+    }
+    if ((this->unk15C >= 0) && (Flags_GetSwitch(play, this->unk15C) != 0)) {
```

Comment:
It's not always obvious what actor params are doing, but in this case it's fairly easy because we can see unk15C (PARAM7F) is fed directly into function `Flags_GetSwitch`. This means it's a switch flag.

A switch flag is a binary boolean flag that signals that something has happened in the world and should be recorded until the next 3-day cycle reset, this can be anything, it's up to the actor: a door has been unlocked, the player has spoken to an actor already, a chest was opened, etc. In this case, if the switch flag is set, this actor is to stop spawning and self-remove itself (this is garo spawner, right? so it won't re-spawn the same garo after you have defeated it I bet)

you can name 15C as `this->switchFlags` and the param as `GET_SWITCHFLAGS`